c/polyjuice.h

```diff
@@ -788,66 +871,72 @@ int load_globals(gw_context_t* ctx, uint32_t to_id) {
+  {
     /* read creator account id and do some checking */
     memcpy(&g_creator_account_id, raw_args_seg.ptr + 32, sizeof(uint32_t));
     int ret = load_account_script(ctx,
                                   g_creator_account_id,
                                   creator_script_buffer,
                                   GW_MAX_SCRIPT_SIZE,
                                   &creator_script_seg);
-    if (ret != 0) {
+    if (ret != 0)
+    {
       return ret;
     }
     mol_seg_t creator_code_hash_seg = MolReader_Script_get_code_hash(&creator_script_seg);
     mol_seg_t creator_hash_type_seg = MolReader_Script_get_hash_type(&creator_script_seg);
     mol_seg_t creator_args_seg = MolReader_Script_get_args(&creator_script_seg);
     creator_raw_args_seg = MolReader_Bytes_raw_bytes(&creator_args_seg);
-    if (memcmp(creator_code_hash_seg.ptr, code_hash_seg.ptr, 32) != 0
-        || *creator_hash_type_seg.ptr != *hash_type_seg.ptr
+    if (memcmp(creator_code_hash_seg.ptr, code_hash_seg.ptr, 32) != 0 || *creator_hash_type_seg.ptr != *hash_type_seg.ptr
         /* compare rollup_script_hash */
-        || memcmp(creator_raw_args_seg.ptr, raw_args_seg.ptr, 32) != 0
-        || creator_raw_args_seg.size != CREATOR_SCRIPT_ARGS_LEN) {
+        || memcmp(creator_raw_args_seg.ptr, raw_args_seg.ptr, 32) != 0 || creator_raw_args_seg.size != CREATOR_SCRIPT_ARGS_LEN)
+    {
       debug_print_int("invalid creator account id in normal contract account script args",
                       g_creator_account_id);
       return FATAL_POLYJUICE;
     }
-  } else {
+  }
+  else
+  {
     debug_print_data("invalid to account script args", raw_args_seg.ptr, raw_args_seg.size);
     return FATAL_POLYJUICE;
   }
-  /** read g_compatible_chain_id from Godwoken RollupConfig */
+  /** read g_chain_id from Godwoken RollupConfig */
   mol_seg_t rollup_config_seg;
   rollup_config_seg.ptr = ctx->rollup_config;
   rollup_config_seg.size = ctx->rollup_config_size;
-  mol_seg_t id_u32_seg = MolReader_RollupConfig_get_compatible_chain_id(&rollup_config_seg);
-  memcpy(&g_compatible_chain_id, id_u32_seg.ptr, id_u32_seg.size);
-  debug_print_int("compatible_chain_id", g_compatible_chain_id);
+  mol_seg_t id_u64_seg = MolReader_RollupConfig_get_chain_id(&rollup_config_seg);
+  memcpy(&g_chain_id, id_u64_seg.ptr, id_u64_seg.size);
+  debug_print_int("chain_id", g_chain_id);
   debug_print_int("creator_account_id", g_creator_account_id);
 
   /** read rollup_script_hash and g_sudt_id from creator account */
   memcpy(g_rollup_script_hash, creator_raw_args_seg.ptr, 32);
   memcpy(&g_sudt_id, creator_raw_args_seg.ptr + 32, sizeof(uint32_t));
   debug_print_data("g_rollup_script_hash", g_rollup_script_hash, 32);
   debug_print_int("g_sudt_id", g_sudt_id);
-  /** read g_eth_addr_reg_id from creator_script_args[36..40] */
-  memcpy(&g_eth_addr_reg_id, creator_raw_args_seg.ptr + 36, sizeof(uint32_t));
+  g_eth_addr_reg_id = GW_DEFAULT_ETH_REGISTRY_ACCOUNT_ID;
```

Comment:
If GW_DEFAULT_ETH_REGISTRY_ACCOUNT_ID is used, then we don't need g_eth_addr_reg_id now.